pkg/controller/drain/drain_controller.go

```diff
@@ -343,6 +304,60 @@ func (ctrl *Controller) syncNode(key string) error {
 	return nil
 }
 
+func (ctrl *Controller) drainNode(node *corev1.Node, drainer *drain.Helper) error {
+	isOngoingDrain := false
+	var duration time.Duration
+
+	for k, v := range ctrl.ongoingDrains {
+		if k != node.Name {
+			continue
+		}
+		isOngoingDrain = true
+		duration = time.Now().Sub(v)
+		glog.Infof("Previous node drain found. Drain has been going on for %v hours", duration.Hours())
+		if duration > drainTimeoutDuration {
+			// TODO right now the daemon will alert to match previous behaviour. Consider having controller do so instead.
+			glog.Errorf("node %s: drain exceeded timeout: %v. Will continue to retry.", node.Name, drainTimeoutDuration)
+			ctrlcommon.MCCDrainErr.Set(1)
+		}
+		break
+	}
+
+	if !isOngoingDrain {
+		ctrl.logNode(node, "cordoning")
+		// perform cordon
+		if err := ctrl.cordonOrUncordonNode(true, node, drainer); err != nil {
+			ctrlcommon.MCCDrainErr.Set(1)
```

Comment:
Hmm, should we be alerting here? It is possible this is temporary and resolves quickly

Reply:
That's a good point. We probably don't want to alert right there.